### DISCUSSION TITLE
github identifies some godot script as a lang named "GAP"

### DISCUSSION THREAD

**Discussion by lin-calvin** [community/community#44976] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

this is a example repo:https://github.com/vcc-chat/vcc_godot
all of the code here is in gdscript but github says it is gap

**Comment by airtower-luna** [comment, marked as answer]
GitHub uses Linguist to detect the language. Check the troubleshooting guide there: https://github.com/github/linguist/blob/master/docs/troubleshooting.md That should help you figure out if its a bug in Linguist (which you could then report), or something you can fix in your repository.

**Comment by dangerousl** [comment]
Adding onto the answer posted previously, there appears to be some collision that occurs between [GAP](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L2091-L2097) and [GDScript ](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L2121-L2125)in linguist.
You can fix this by committing a `.gitattributes` file to your repo containing the following line:

```
*.gd linguist-language=gdscript
```

This will override all `.gd` extensions to use GDScript instead. [You can learn more about overrides in the linguist docs, too.](https://github.com/github/linguist/blob/master/docs/overrides.md)
